### ISSUE TITLE
batch inference return NaNs

### ISSUE THREAD

**Issue by camperr** [NVIDIA/TensorRT#1550]
## Description
I try to depoly albert-finetune model follow the path pytorch->onnx->tensorrt->triton server. When I input line by line, the result is normal, but when I input batch, only the first line is the normal result, and all other lines are nan.

## Environment

**TensorRT Version**:  8.0.1, 8.0.3
**NVIDIA GPU**: 2080Ti
**NVIDIA Driver Version**:  470.63.01 
**CUDA Version**:  11.4
**CUDNN Version**: 
**Operating System**: CentOS
**PyTorch Version (if applicable)**: 1.8
**Tensorrt Image**:nvcr.io/nvidia/tensorrt:21.07-py3,21.09-py3

## Relevant Files
ONNX model 
link: [https://pan.baidu.com/s/1LI9xAaq68KhgdjkXqOVRzQ](url) code: 3reb

## Steps To Reproduce

There are two ways to reproduce, one is just run trtexec command in tensorrt image
`trtexec --explicitBatch --onnx=albert.onnx \
    --minShapes=\'input_ids\':1x512,\'att_masks\':1x512,\'token_ids\':1x512,\'source_ids\':1x1,\'intent_ids\':1x1 \
    --optShapes=\'input_ids\':4x512,\'att_masks\':4x512,\'token_ids\':4x512,\'source_ids\':4x1,\'intent_ids\':4x1 \
    --maxShapes=\'input_ids\':8x512,\'att_masks\':8x512,\'token_ids\':8x512,\'source_ids\':8x1,\'intent_ids\':8x1 \
    --inputIOFormats=int32:chw \
    --outputIOFormats=fp32:chw,fp32:chw,int32:chw \
    --saveEngine=albert.trt \
    --workspace=4196 \
    --verbose \
    --dumpOutput`

Output:
[10/11/2021-13:08:44] [I] Output Tensors:
[10/11/2021-13:08:44] [I] preds: (4)
[10/11/2021-13:08:44] [I] 2 -1 -1 -1
[10/11/2021-13:08:44] [I] probs: (4x4)
[10/11/2021-13:08:44] [I] 0.130603 0.245327 0.320006 0.304064 nan nan nan nan nan nan nan nan nan nan nan nan
[10/11/2021-13:08:44] [I] logits: (4x4)
[10/11/2021-13:08:44] [I] 0.00684099 0.637267 0.903015 0.851913 nan nan nan nan nan nan nan nan nan nan nan nan

When i add --fp16 to trtexec, result would be all NaN
[10/11/2021-12:49:21] [I] Output Tensors:
[10/11/2021-12:49:21] [I] preds: (4)
[10/11/2021-12:49:21] [I] -1 -1 -1 -1
[10/11/2021-12:49:21] [I] probs: (4x4)
[10/11/2021-12:49:21] [I] nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan
[10/11/2021-12:49:21] [I] logits: (4x4)
[10/11/2021-12:49:21] [I] nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan

--------------------------------------------------------------
second way to reproduce, 
1. deploy with triton image nvcr.io/nvidia/tritonserver:21.09-py3

`docker run -itd --rm -p8197:8000 -v/home/zhangnan/triton/models:/models --env NVIDIA_VISIBLE_DEVICES=1 63e21e0cb5df ./bin/tritonserver --model-repository=/models --allow-http=true --log-verbose=4`

config.pbtxt
```
name: "ms_sim"
platform: "tensorrt_plan"
max_batch_size: 8
default_model_filename: "model.plan"
input [
    {
        name: "input_ids"
        data_type: TYPE_INT32
        dims: [ 512 ]
    },
    {
        name: "att_masks"
        data_type: TYPE_INT32
        dims: [ 512 ]
    },
    {
        name: "token_ids"
        data_type: TYPE_INT32
        dims: [ 512 ]
    },
    {
        name: "source_ids"
        data_type: TYPE_INT32
        dims: [ 1 ]
    },
    {
        name: "intent_ids"
        data_type: TYPE_INT32
        dims: [ 1 ]
    }
]
output [
    {
        name: "logits"
        data_type: TYPE_FP32
        dims: [ 4 ]
    },
    {
        name: "probs"
        data_type: TYPE_FP32
        dims: [ 4 ]
    },
    {
        name: "preds"
        data_type: TYPE_INT32
        dims: [ 1 ]
        reshape: { shape: [ ] }
    }
]

dynamic_batching {
    preferred_batch_size: [ 4, 8 ]
    max_queue_delay_microseconds: 100
}

instance_group [
    {
        count: 1
        kind: KIND_GPU
        gpus: [ 0 ]
    }
]
```
2. And send requests with tritonclient
```
import numpy as np
import sys
import json
from tritonclient.http import InferenceServerClient, InferInput, InferRequestedOutput
from tritonclient.utils import InferenceServerException
import time
import re

class SimService:
    def __init__(self):
        self.url = '127.0.0.1:8197'
        self.triton_client =InferenceServerClient(url=self.url, verbose=False, concurrency=8)
        self.model_name ="ms_sim"

    def infer(self):
        inputs = []
        outputs = []
        batch = 4
        inputs.append(InferInput('input_ids', [batch, 512], "INT32"))
        inputs.append(InferInput('att_masks', [batch, 512], "INT32"))
        inputs.append(InferInput('token_ids', [batch, 512], "INT32"))
        inputs.append(InferInput('source_ids', [batch, 1], "INT32"))
        inputs.append(InferInput('intent_ids', [batch, 1], "INT32"))

        # Initialize the data
        inputs[0].set_data_from_numpy(np.ones((batch, 512), dtype=np.int32), binary_data=False)
        inputs[1].set_data_from_numpy(np.ones((batch, 512), dtype=np.int32), binary_data=False)
        inputs[2].set_data_from_numpy(np.ones((batch, 512), dtype=np.int32), binary_data=False)
        inputs[3].set_data_from_numpy(np.ones((batch, 1), dtype=np.int32), binary_data=False)
        inputs[4].set_data_from_numpy(np.ones((batch, 1), dtype=np.int32), binary_data=False)

        outputs.append(InferRequestedOutput('probs', binary_data=True))
        outputs.append(InferRequestedOutput('preds', binary_data=True))

        result = self.triton_client.infer(self.model_name, inputs, outputs=outputs, headers=None)
        print(result.as_numpy('probs'))
        print(result.as_numpy('preds'))

if __name__ == '__main__':
    ss = SimService()
    ss.infer()
```
When reproduce with the second way, fp32 give normal result** and fp16 give one line normal and other line NaN

**Comment by ttyio** [comment]
Sorry @camperr for the delay response, the file is expired, could you share again? thanks

**Comment by camperr** [comment]
> 

link: https://pan.baidu.com/s/1SxWapC6jr9R6CtMNqxgDgA code: meca

**Comment by ttyio** [comment]
Thanks @camperr , could you try 8.2 release? It is not NaN in my side, but I am using RTX8000:

[01/07/2022-09:59:33] [I] Output Tensors:
[01/07/2022-09:59:33] [I] preds: (4)
[01/07/2022-09:59:33] [I] 2 2 1 1
[01/07/2022-09:59:33] [I] probs: (4x4)
[01/07/2022-09:59:33] [I] 0.165891 0.265296 0.290839 0.277975 0.192599 0.279321 0.303641 0.22444 0.201888 0.279092 0.260206 0.258815 0.141423 0.335987 0.289609 0.23298
[01/07/2022-09:59:33] [I] logits: (4x4)
[01/07/2022-09:59:33] [I] 0.157531 0.627048 0.718972 0.673734 0.294113 0.665865 0.74935 0.44711 0.206773 0.530601 0.460534 0.455175 -0.177234 0.68808 0.539539 0.321962

**Comment by ttyio** [comment]
Closing since no activity for more than 3 weeks, thanks!
